pkg/config/loader_test.go

```diff
@@ -0,0 +1,305 @@
+package config
+
+import (
+	"os"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+
+	autoheal "github.com/openshift/autoheal/pkg/apis/autoheal/v1alpha2"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestFile(t *testing.T) {
+	l := NewLoader()
+	l.File("test_file")
+	if len(l.files) != 1 {
+		t.Fail()
+	}
+
+	if l.files[0] != "test_file" {
+		t.Fail()
+	}
+}
+
+func TestFiles(t *testing.T) {
+	l := NewLoader()
+	l.Files([]string{"test_file0", "test_file1", "test_file2"})
+	if len(l.files) != 3 {
+		t.Fail()
+	}
+
+	for idx, file_name := range l.files {
+		if file_name != ("test_file" + strconv.Itoa(idx)) {
+			t.Fail()
+		}
+	}
```

Comment:
Same, try to avoid using the private fields. In this particular case you can do the same putting different parameters in different files, and then checking that all of them have been loaded. For example, `test_file0` can contain this:

```yaml
awx:
  address: https://my-awx.example.com
```

Then `test_file1` can contain this:

```yaml
awx:
  proxy: http://my-proxy.example.com:3128
```

Then you can load both files and check that both `awx.address` and `awx.proxy` are populated and that they have the expected values.